src/docker/composeappengine.cc

```diff
@@ -151,6 +151,15 @@ AppEngine::Result ComposeAppEngine::run(const App& app) {
   return result;
 }
 
+void ComposeAppEngine::stop(const App& app) {
+  try {
+    const auto root_dir{appRoot(app)};
+    exec(compose_ + "down", "failed to bring App down", boost::process::start_dir = root_dir);
+  } catch (const std::exception& exc) {
+    LOG_ERROR << "docker-compose was unable to bring down: " << exc.what();
+  }
+}
```

Comment:
this is 99% the same as `remove`. You can remove most of `remove`'s logic and just call this now.

Reply:
It doesn't make sense to reuse `stop` from `remove` because we need to put try/catch into `remove` anyway (remove_all can throw exc). So, what's the point to run `this->stop()` if we can run `exec(compose down)` directly.